app/jinja_filters.py

```diff
@@ -43,6 +43,23 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(dates):
+    """ This function format_cond_date accepts two dates, if a date is submitted that is used and is available,
+    if not the metadata date is made available The format (example) in the json is
+    {{[answers.date_from, exercise.start_date]|format_conditional_date}} """
+
+    if dates[0]:
+        date = dates[0]
+    else:
+        date = dates[1]
+
```

Comment:
Existence of date[0] checked, if no date[0], date is set to value of date[1]. Date is then checked and if None the exception is thrown. Thus the existence check on date[1] is carried out implicitly. A second check could be added, but this would add little value.